ui/lib/core/addon/modifiers/code-mirror.js

```diff
@@ -26,8 +27,18 @@ export default class CodeMirrorModifier extends Modifier {
     } else {
       // this hook also fires any time there is a change to tracked state
       this._editor.setOption('readOnly', namedArgs.readOnly);
-      if (namedArgs.content && this._editor.getValue() !== namedArgs.content) {
-        this._editor.setValue(namedArgs.content);
+      let value = this._editor.getValue();
+      let content = namedArgs.content;
+      if (!content) return;
+      try {
+        // we use the stringify helper so we do not flatten the json object
```

Comment:
it might be confusing why we're parsing and then stringifying, can we add back the comment that says we `parse` first to remove superfluous whitespace so that we can just compare key/value pairs instead of the line breaks and extra spaces?